### PR TITLE
fix: DEV-3337: add storage_filename to filters (contains and exact) 

### PR DESCRIPTION
**Pull request by chudomozg**
Above all, thank you for label-studio. I would like to point out that I am pushing to opensource project first time. So please, don't be too harsh if i do something wrong.

There is issue [#3337](https://github.com/heartexlabs/label-studio/issues/3337) about the inability to filter by storage_name. I added ability to filtering by storage_filename. I've used [django-queryable-properties](https://github.com/W1ldPo1nter/django-queryable-properties) library. So, now we can use any properties for filters. I have only implemented a few filters: icontains and exact. Because there are some trouble with __isnull filter and regex.

### CODE REVIEW

File: label_studio/tasks/models.py [hunk 1]
```diff
@@ -45,9 +45,10 @@ class Task(TaskMixin, models.Model):
     """ Business tasks from project
     """
     id = models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID', db_index=True)
-    data = JSONField('data', null=False, help_text='User imported or uploaded data for a task. Data is formatted according to '
-                                                   'the project label config. You can find examples of data for your project '
-                                                   'on the Import page in the Label Studio Data Manager UI.')
+    data = JSONField('data', null=False,
```
Comment (makseq) [inline label_studio/tasks/models.py:48]: please don't change lines that are not related to your PR

**Review by makseq — CHANGES_REQUESTED** [review]

**Comment by chudomozg** [comment]
I've found some problem in my code when I was testing it. I will fix it and open new merge request.
